README.md

```diff
@@ -4,27 +4,242 @@
+   * [OpenSRP KIP application](https://github.com/OpenSRP/opensrp-client-kip)
+   * [OpenSRP Growth monitoring library](https://github.com/OpenSRP/opensrp-client-growth-monitoring)
+   * [OpenSRP Immunization library](https://github.com/OpenSRP/opensrp-client-immunization)
+   * [OpenSRP Native Form library](https://github.com/OpenSRP/opensrp-client-native-form)
+
+
+# Main Functions
+
+## 1. Security
+
+Security is provided in the following:
+   * Network - It supports ssl in cases where the backend server is connected through a `https` url
+   * Data access - Only registered providers can be able to view and manipulate records
+   * Data encryption - The database on the android client is encrypted
+
+The security classes can be found in `org.smartregister.ssl`
+
+## 2. Data management
+
+This app provides data management for the following:
+   * Alerts
+   * Eligible couples
+   * Reports
+   * Services Provided
+   * Settings
+   * Name-Value pairs
+   * Timeline events
+   * Children
+   * Details
+   * Events
+   * Form data
+   * Form Versions
+   * Image Locations
+   * Mothers
+   * Reports
+   * Device Users
+   * Connection configurations
+
+The data management classes can be found in `org.smartregister.repository`
+
+## 3. Networking
+
+This app provides the following networking capabilities:
+   * SSL Connection helper
+   * Asynchronous networking classes
+   * Synchronous networking classes
+   * Network status detection
+   * Thread safe connections
+   * GZip encoding and decoding capabilities
+   * Session management
+   * Client-Server time synchronization
+   * User authentication
+
+The networking classes can be found in:
+   * `org.smartregister.service`
+   * `org.smartregister.util`
+   * `org.smartregister.client`
+
+
+## 4. Domain Objects
+
+This app provides the following domain objects:
+   * Database Domain
+      * Address
+      * Base Data Object
+      * Base Entity
+      * Client
+      * Column
+      * Column Attribute
+      * Event
+      * Filter Type
+      * Obs
+      * Query
+   * Form Domain
+      * Field Overrides
+      * Form Data
+      * Form Field
+      * Form Instance
+      * Form Submission
+   * Global Domain
+      * Alert
+      * ANM (Health Services Provider)
+      * Child
+      * Eligible Couple
+      * Form Definition Version
+      * Mother
+      * Photo
+      * Profile Image
+      * Report
+      * Montly Report
+      * Http response
+      * Service Provided
+      * Timeline Event
+
+The domain object classes can be found in `org.smartregister.domain`
+
+## 5. Sync
+
+This app provides the following sync capabilities:
+
+   * Periodic syncing based on network connection
+
+The sync classes can be found in `org.smartregister.sync`
+
+## 6. Utitiles
+
+This app provides the following utilties:
+   * Improved Image caching
+   * Data caching
+   * File storage utilities
+   * Float conversion utility
+   * Form Submission builders
```

Comment:
same thing as above, let's use w/consistent all words capitalized or consistent only first word capitalized

Reply:
Some words describe models while others form the model name(s)

Reply:
cool, are models classes? Like in the code is this `FormSubmission`? What about writing like that in here

Reply:
Yea, but since we are trying to explain the functionality or direct real world representatives, I was trying as much as possible make it easily understable